### ISSUE TITLE
Module parameter inheritance is only done at import time

### ISSUE THREAD

**Issue by fabianegli** [nextflow-io/nextflow#1984]
The DSL2 documentation is not explicit about the module parameter inheritance when it comes to procedure. Inheritance is only available at the time of inclusion, not later on.

Relevant docs: https://www.nextflow.io/docs/latest/dsl2.html#module-parameters

Note that in the example below we see a [Spanglish](https://en.wikipedia.org/wiki/Spanglish) sentence.

```
$ cat module 
params.foo = 'Hello'
params.bar = 'world!'

def sayHello() {
    println "$params.foo $params.bar"
}

$ cat wf.nf 
nextflow.enable.dsl=2

params.foo = 'Hola'

include {sayHello} from './module'

params.bar = 'Mundo'

workflow {
    sayHello()
}

$ nextflow wf.nf 
N E X T F L O W  ~  version 20.10.0
Launching `wf.nf` [naughty_lavoisier] - revision: fae2d88139
Hola world!
```

**Comment by andersgs** [comment]
I just got caught on this one.

**Comment by pditommaso** [comment]
This was solved by #1985. Thanks for reporting the problem
